project_forecast_line_priority/models/project_task.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2024 Therp BV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from datetime import timedelta
+
+from odoo import fields, models
+
+
+class ProjectTask(models.Model):
+    _inherit = "project.task"
+
+    def write(self, vals):
+        """Set forecast_date_planned_end based on priority"""
+        # If user sets end date manually
+        if "forecast_date_planned_end" in vals or "date_deadline" in vals:
+            return super().write(vals)
+        if "priority" not in vals:
+            return super().write(vals)
+        priority = vals.get("priority", -5)
+        for this in self:
+            # date deadline is set, so ignore this one
+            if this.date_deadline:
+                continue
+            # if priority is not set, do nothing
+            if int(priority) < 0:
+                continue
+            forecast_date_planned_end = self._get_forecast_date_planned(
+                priority=priority
+            )
+            if not forecast_date_planned_end:
+                continue
+            vals["forecast_date_planned_end"] = forecast_date_planned_end
```

Comment:
vals is updated here for all records, even the records that do not satisfy the conditions for instance the ones that do have a date_deadline.

Reply:
Agreed, I intended to change that, though it is not clear to me how to edit vals for some records, and not for others. I'll just edit priority for a recordset on shell, and check what vals contain in that case.

Reply:
Maybe things will be easier is just starting with `result = super().write(vals)`. Then when priority was in vals, update just the records that have no deadline set according to priority. Something like:
```
result = super().write(vals)
if "priority" in vals:
    no_deadline_tasks = self.filtered(r: not r.date_deadline)
    for task in no_deadline_tasks:
            forecast_date_planned_end = task._get_forecast_date_planned(
                priority=priority
            )
            if forecast_date_planned_end:
                super(ProjectTask, task).write(
                   {"forecast_date_planned_end": forecast_date_planned_end}
                )
return result
```
Note that in the present code _get_forecast_date_planned is called on self which might result in an Exception.

Reply:
Actually I am about to push a new commit about this one, with a few lines changed compared to previous version. Will ping you